### DISCUSSION TITLE
Image optimizations for background images

### DISCUSSION THREAD

**Discussion by SogoCZE** [vercel/next.js#18357] [Help] [answered]
Edit: this is now supported: https://github.com/vercel/next.js/discussions/18357#discussioncomment-662118

**Comment by mercteil** [comment]
Yep, just facing the same "issue". Specifically images that are used to cover/contain the background are set via css. Not entirely clear to me how to use `next/image` here.

**Comment by mxmtsk** [comment]
I usually do something like this. Hope this helps for some of your use cases. I have yet to find a downside to this. :)

```jsx
<div className="relative">
    <Image
      layout="fill"
      className="object-center object-cover pointer-events-none"
      src={backgroundImage}
      alt={title}
    />
   <div className="relative z-1">
       my content
   </div>
</div>
```

Demo with plain HTML:
https://jsfiddle.net/mxmtsk/cdh9vxnm/

**Reply by Luisdv93** [reply]
It was so hard to find something related to this problem. This solution worked! At least so far, I'll keep working with and see it suits my specific use-case. 😁

Thank you

**Reply by rwacarter** [reply]
This works flawlessly from a display point of view. The issue is semantics - an `<img />` and `background-image` represent different things, either being part of the content or pure decoration, respectively. So although you can use CSS to make an `<img />` behave similarly to `background-image`, I still think there's an argument for a solution that uses an actual `background-image` on an element.

**Reply by mxmtsk** [reply]
@rwacarter My solution was also just an idea while real bg image support is not present. :) I'm no accessibility expert, but if that's a requirement for you, adding `role="presentation"` should treat the image right for screen readers at least, no?

**Reply by chanhman** [reply]
From an accessibility stand point, adding an empty `alt=""` will tell screen readers that the image is decoration and it should skip it.

**Reply by zahidur59** [reply]
Haven't tried it yet have a question. We do things like we can do with background images. Like "Cover"
 right?
 
EDIT:  Never-mind. Just saw the post bellow.

**Reply by bn-l** [reply]
NB: this uses object-fit which doesn't work on ie 11 (but there's a polyfill)

**Reply by mledwards** [reply]
I add pointer-events:none, then you don't need to worry about z indexing with future changes.

**Reply by neaucode** [reply]
another point here is to use `clip-path: inset(0)` on a wrapper so that you can have multiple elements with fixed image backgrounds
https://github.com/vercel/next.js/discussions/18357#discussioncomment-1941014

**Comment by guitheengineer** [comment, marked as answer]
`next/image` now supports the prop `objectFit="cover"` which can make the component work as a background image

official demo:
https://image-component.nextjs.gallery/background

docs: 
https://nextjs.org/docs/api-reference/next/image#objectfit

**Reply by huzaifah050** [reply]
This looks like a good hack but it has its limitations. Want to implement these css and I am having a hard time. `  background-position: center;
  background-color: red;
  background-blend-mode: multiply;`

**Reply by guitheengineer** [reply]
You can use `objectPosition="center"` to centralize the image.

For background-color I don't know why if you're using an Image though, but it should be possible to just wrap in a div and style it.

**Reply by huzaifah050** [reply]
What of the background-blend-mode ?

**Reply by rwacarter** [reply]
This again goes back to the point I made on `mxmtsk`'s comment above. I know there are lots of hacks and workarounds in CSS to make an `<img>` look like a `background-image`, but they are fundamentally different things, and as such there will always be these semantic and edge-case issues. Using the same performance optimization logic, lazy loading, etc and applying it to `background-image` is still something sorely missing from NextJS.

**Reply by zwily** [reply]
What if you want a repeating background?

**Reply by RobSchilderr** [reply]
> You can use `objectPosition="center"` to centralize the image.
> 
> For background-color I don't know why if you're using an Image though, but it should be possible to just wrap in a div and style it.

Yet you don't have responsive background positioning which is still necessary for a dynamic header

**Reply by Amirmohammadkh** [reply]
How to make this image repeat in the background?

**Comment by sanderkooger** [comment]
Hey all, being able to generate a URL for background components would be amazing, But it would not stop there. It would allow us to use sized OG images etc where ever needed. I think a function like this would be epic. 

`const imageSettings = {height= heightInPX, width= WidthInPX, quality={1/100}, ETC};`

`const ogImage = useNextImage ('https://cdnthingie/serverthingie.com/imageName.jpeg', imageSettings);`


This should use the same mechanics as next image.

**Reply by sanderkooger** [reply]
I have done some research, looks liek a method of doing this woudl be building your own API within your nextjs app.... an SSR only page that does 
https://monoppa.com/blog/image-transformation-in-nextjs

**Comment by Eftychiou** [comment]
Hello team, i am having tough time finding the solution to the parallax effect that is achieved using a backround-image(url...) and background-position:fixed. Have you guy found a work around to that ?

These 3 css lines achieved that 
  //   background: url('/images/banners/banner5.jpg') no-repeat center ;
  //   background-size: cover ;
  //  background-attachment: fixed ; 

So how do we do it with next/image if we can't use the  css background ?

**Reply by RobSchilderr** [reply]
Good point @Eftychiou. Looking for an answer too. I really need to use the background position, but also would like to lazy load.

**Reply by MartijnHols** [reply]
You can use position:fixed with clip-path on the parent container: https://stackoverflow.com/questions/33550450/how-to-use-css-background-attachment-in-img-tag/68466386#68466386

**Reply by neaucode** [reply]
@MartijnHols this entirely solves the issue, thank you so much, this allows multiple elements with fixed backgrounds

**Comment by jserrao** [comment]
My use case was that I needed background images that could have other imagery/text layered on top of them. This is a pretty standard use case but hard to achieve with the `next/images` and other similar approaches that force images into the foreground.

I just basically built the primitive I wish Next had - it uses styled-components but you could easily revert to CSS if you prefer. Hope it helps others:

```jsx
import Image from "next/image"
import styled from 'styled-components'

const Container = styled.section`
  align-items: ${ props => props.alignItems };
  display: flex;
  justify-content: ${ props => props.justifyContent };
  position: relative;
  height: ${ props => props.height };
  width: ${ props => props.width };
`;

const InnerContainer = styled.div`
  z-index: 10;
`

/**
 * <BgImage>
 *
 * The new next/image optimization setup handles background images oddly
 * It requires they be foreground images placed inside of a container
 * This component abstracts that logic away for better DX
 * 
 * You can layer text and imagery on top of the background image
 * All you have to do is pass that text or imagery into this component
 * 
 * Note: all images get processed through Webpack so you must import! 
 * No absolute URLs as they will break during site generation
 *
 * @param { string } alignItems - vertical alignment of inner content
 * @param { string } imgalt - text description of the image
 * @param { string } imgsrc - url of the image, should be a JS module import
 * @param { string } justifyContent - horizontal alignment of inner content
 * @param { number } height - how tall the background image should be (default: 50vh)
 * @param { number } width - how wide image should be (default: 100%)
 */
const BgImage = ({ 
  alignItems = 'center',
  children,
  imgalt = 'Background Image',
  imgsrc,
  height = '50vh',
  justifyContent = 'center',
  width = '100%'
}) => {

  return(
    <Container
      alignItems={ alignItems }
      height={ height }
      justifyContent={ justifyContent }
      width={ width }
    >
      <Image
        alt={ imgalt }
        src={ imgsrc }
        layout="fill"
        objectFit="cover"
        quality={ 80 }
      />
      { children && 
        <InnerContainer>
          { children }
        </InnerContainer>
      }
    </Container>
  )
}

export default BgImage
```

**Comment by niklasgrewe** [comment]
What about seo? can I make sure that an image is not indexed by Google when using the above methods? Normally you use `background-image` in such cases, right? 

I want to optimize my images, but don't want them to be indexed by google. How can I solve this?

**Reply by jserrao** [reply]
https://developers.google.com/search/docs/advanced/crawling/prevent-images-on-your-page

**Comment by cazterk** [comment]
my settings 
tsx =>  `     <div
                    className={styles.card__mainImage}
                    >
                    <Image
                      src={image source} 
                      objectFit="cover"
                      layout="fill"
                      quality={65}
                    />
                  </div>`
css => `.card__mainImage{
      flex: 0 0 290px;
      height: 190px;
      position: relative;
      background-size: cover;
      background-repeat: no-repeat;
      background-position: center center;
      border-radius: 1px;
    }`

**Comment by sovetski** [comment]
The selected "correct answer" is not really the correct one for me... How about repeated background images? https://www.reddit.com/r/tailwindcss/comments/pdw09p/background_repeat_nextjs_image_with_tailwind_css/

**Comment by dnlaviv** [comment]
were you guys able to apply a parallax effect to an image component used as background? 
I'm struggling to get the scroll effect since I can't find a solution equivalent to backgound-size property in object-fit approach

**Comment by grigor-minasyan** [comment]
I also came across this issue, and I think the best way would probably be to just not use next/image in these cases, we're not required to use it. We can just optimize the image ourselves and use regular URLs for the background image.
